### PR TITLE
Update name and metadata for the extension

### PR DESCRIPTION
**Pull request by larohra**
- Updated name of the extension
- Added a readme for the public page
- Added an icon

### CODE REVIEW

File: metadata/extension-readme.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# ADO Extension for Defender for DevOps
+
+An extension for Azure DevOps that allows [Microsoft Defender for DevOps](https://aka.ms/DfDdocs) to extract metadata from pipelines, such as containers digest ID, container image name etc. That information will then be used within [Microsoft Defender for Cloud](https://aka.ms/mdc) to connect DevOps entities with their related resources running multi-cloud (Azure, AWS, GCP, on-premise).
```
Comment (sethRait) [inline metadata/extension-readme.md]: Grammar: "containers digest ID" should be either "container digest IDs" or "a container's digest ID"

Comment (sethRait) [inline metadata/extension-readme.md]: Readability: "to connect DevOps entities with their related resources running multi-cloud (Azure, AWS, GCP, on-premise)."
Consider rewording this. It wasn't clear to me exactly what this meant until I read it twice, so I think customers may be confused.
"...to connect DevOps entities with their related cloud resources (Azure, AWS, GCP, on-premises) " might be better. You don't need to say "running multi-cloud" if you include that info in the parentheses anyway.

File: metadata/extension-readme.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## Dependencies
+
+The task currently depends on [Docker](https://docs.docker.com/get-started/overview/).
```
Comment (sethRait) [inline metadata/extension-readme.md]: Readability: The -> This
also append this sentence with "being installed on the build agent" (or similar

File: metadata/extension-readme.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+Microsoft-hosted build agents ship with an included list of software. To see if your agent image comes with these pre-installed, [see here](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software).
+
+To install Docker manually, please follow the steps listed on their [docs page](https://docs.docker.com/engine/install/).
+
+## View Results
+
+To better view the results of the code to cloud, open Microsoft Defender for Cloud in your Azure portal.
```
Comment (sethRait) [inline metadata/extension-readme.md]: "To better review the results" implies that you've already given a way of viewing results which is not as good, but you haven't. Reword this. Also, "the code to cloud" is an awkward term. Maybe "of the assessment" or similar.

File: vss-extension.json [hunk 4]
```diff
@@ -1,10 +1,10 @@
 {
     "manifestVersion": 1,
-    "id": "DefenderForDevops-PipelineDecorators",
+    "id": "ms-dfd-code-to-cloud",
```
Comment (sethRait) [inline vss-extension.json:3]: If we change this ID, we will need an accompanying code change on the backend, fyi

File: vss-extension.json [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
     "manifestVersion": 1,
-    "id": "DefenderForDevops-PipelineDecorators",
+    "id": "ms-dfd-code-to-cloud",
     "publisher": "ms-securitydevops",
-    "version": "0.0.2",
-    "name": "DefenderForDevops-PipelineDecorators",
-    "description": "ADO Pipeline Decorator Extension for Defender for Devops",
+    "version": "0.0.3",
+    "name": "Microsoft Defender for DevOps - Code to cloud",
+    "description": "Code to cloud extension for Microsoft Defender for DevOps",
```
Comment (sethRait) [inline vss-extension.json]: This description should still include the fact that this is an extension for ADO.

**Review by sethRait — CHANGES_REQUESTED** [review]

File: vss-extension.json (hunk 4 above)
Comment (larohra) [reply vss-extension.json:3]: Yup agreed. The only backend I am aware of is the enablement process in dfd-rp. Is there anything else? That should be an easy change right?

Comment (sethRait) [reply vss-extension.json:3]: That's all, and this is fine just wanted to bring it up

**Review by sethRait — APPROVED** [review]

**Pull request merged**